darwin/fixed.m

```diff
@@ -0,0 +1,201 @@
+// 15 august 2015
+#import "uipriv_darwin.h"
+
+// TODO hiding all stretchy controls still hugs trailing edge
+
+@interface fixedChild : NSObject
+@property uiControl *c;
+@property int x;
+@property int y;
+- (NSView *)view;
+@end
+
+@interface fixedView : NSView {
+	uiFixed *b;
+	NSMutableArray *children;
+}
+- (id)initFixed:(uiFixed *)bb;
+- (bool)isFlipped;
+- (void)onDestroy;
+- (void)syncEnableStates:(int)enabled;
+- (void)append:(uiControl *)c x:(int)x y:(int)y;
+- (void)move:(uiControl *)c x:(int)x y:(int)y;
+- (void)reloadPositions;
+@end
+
+struct uiFixed {
+	uiDarwinControl c;
+	fixedView *view;
+};
+
+@implementation fixedChild
+
+- (NSView *)view
+{
+	return (NSView *) uiControlHandle(self.c);
+}
+
+@end
+
+@implementation fixedView
+
+- (id)initFixed:(uiFixed *)bb
+{
+	self = [super initWithFrame:NSZeroRect];
+	self.autoresizingMask = NSViewWidthSizable | NSViewHeightSizable;
```

Comment:
Always use Objective-C method syntax, even for superclass property getters and setters.

Put this line in the if block.

libui uses Auto Layout; `uiprivFixedView` should be controllable with Auto Layout, but its contents don't necessarily have to be. I have to figure out how this would work.

Reply:
I don't think the autoresizingMask is strictly necessary, I think I just added that when trying to debug why the components weren't being sized correctly.